src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.java

```diff
@@ -0,0 +1,114 @@
+import com.google.jenkins.plugins.computeengine.client.ComputeClient;
+import com.google.jenkins.plugins.computeengine.integration.ITUtil.InstanceConfigurationBuilder;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT {
+  private static Logger log =
+      Logger.getLogger(ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.class.getName());
+
+  private static final String MULTIPLE_NUM_EXECUTORS = "2";
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label =
+      getLabel(ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.class);
+  private static Instance instance;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    addClassLogHandler(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    addClassLogHandler(streamHandler, ComputeClient.class.getName());
+
+    cloud.addConfiguration(
+        new InstanceConfigurationBuilder()
+            .startupScript(DEB_JAVA_STARTUP_SCRIPT)
+            .numExecutors(MULTIPLE_NUM_EXECUTORS)
+            .jenkinsLabels(LABEL)
+            .label(label)
+            .createSnapshot(false)
+            .oneShot(false)
+            .template(NULL_TEMPLATE)
+            .build());
+
+    Collection<PlannedNode> planned = cloud.provision(new LabelAtom(LABEL), 2);
+    assertEquals(logs(streamHandler, logOutput), 1, planned.size());
+    planned.iterator().next().future.get();
+
+    instance = client.getInstance(PROJECT_ID, ZONE, planned.iterator().next().displayName);
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    teardownResources(streamHandler, logOutput, client, label, log);
+  }
+
+  @Test
+  public void test1WorkerCreatedFor2ExecutorsStatusRunning() {
+    assertEquals(logs(streamHandler, logOutput), "RUNNING", instance.getStatus());
+  }
+
+  @Test
+  public void test1WorkerCreatedFor2ExecutorsExpectedLogs() {
```

Comment:
Just my personal opinion..

I would never test logs..
It doesynt make much sense in my opinion..

Reply:
Agreed, much better to test the state directly.